Feature/#139 모든 뷰와 뷰모델 연결 완료

Opunundo opened the pull request:
## 📕 Issue Number

Close #139 

## 📙 작업 내역

> 수술 카드 뷰를 위한 별도의 디스플레이 모델을 추가하였습니다.
오늘의 수술 뷰에서 수술 카드 삭제 기능 구현을 위해 TodaysSurgeryViewModel에 extension으로 deleteOperationCard 메서드를 추가하였습니다.


-  수술 수정 기능도 완료하였습니다. 하지만 글자 수가 길면 잘리는 현상이 있네요.... 어쨋든 되긴 됩니다....
실제 수정은 HomeView의 시트에서 이루어지고요. 시트 표시 조건과 인풋모드만 환자 디테일 뷰와 오늘의 수술 뷰에서 바꾸는 방식으로 구현하였습니다. 그리고 각각의 뷰모델에 선택된 환자 id와 수술 id에 대한 값도 추가하였습니다. rootViewModel은 이들 값을 소유하고 있고요. 그래서 rootVM이 소유한 환자 id와 수술 id가 있기 때문에 rootVM에서는 별도의 매개변수 없이 updateOperation을 호출할 수 있습니다.

-> OperationViewModel에 매개변수 받는 버전 updateOperation( ... ) 추가하였습니다.
-> MacRootViewModel에서 수술 수정도 담당합니다.
-> 하위 viewModel들은 수술 시트 트리거만 담당합니다.

- [x] 작업 내역 작성


## 📱 스크린샷

> UI 구현 혹은 화면 변동이 있는 PR이라면 스크린샷 첨부

https://github.com/user-attachments/assets/260db230-f2cc-4238-aab2-1e8843ceee3b

https://github.com/user-attachments/assets/2683e0d9-cc4b-4490-bc47-770c5c45c123



## 📘 작업 유형

- [ ] 신규 기능 추가
- [ ] 버그 수정
- [ ] 리펙토링
- [ ] 문서 업데이트


## 📝 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- 특이 사항 1
HomeViewModel에 updateOperation 관련 메서드가 없습니다. 수정 기능을 위한 뷰모델 기능 정리가 필요해 보입니다.
-> OperationViewModel에 매개변수 받는 버전 updateOperation( ... ) 추가하였습니다.
-> MacRootViewModel에서 수술 수정도 담당합니다.


- 특이 사항 2

<br><br>

Opunundo committed:
feature: 모든 뷰와 뷰모델 연결 완료(수정 기능 제외)

수정 기능은 뷰모델에 없어서 원띵과 협의 후에 추가하겠습니다.

freeskyES reviewed (approved):
크흐,, 고민하고 학습해가는 헤기 정말 멋집니당.!

Opunundo committed:
needs fixing: 수술 수정. 나머진 다 되는데 UI 업데이트가 안 됨.

수술 수정 되는 것까진 로그로 확인하였으나, UI에 반영이 안 됩니다...

Opunundo committed:
feature: 수술 수정 기능 연결 완료

하지만 글자 수가 길면 잘리는 현상이 있네요.... 어쨋든 되긴 됩니다....
실제 수정은 HomeView의 시트에서 이루어지고요. 시트 표시 조건과 인풋모드만 환자 디테일 뷰와 오늘의 수술 뷰에서 바꾸는 방식으로 구현하였습니다. 그리고 각각의 뷰모델에 선택된 환자 id와 수술 id에 대한 값도 추가하였습니다. rootViewModel은 이들 값을 소유하고 있고요. 그래서 rootVM이 소유한 환자 id와 수술 id가 있기 때문에 rootVM에서는 별도의 매개변수 없이 updateOperation을 호출할 수 있습니다.

Opunundo committed:
fix: 대문자 수정
